### PR TITLE
Advanced state sensors

### PR DESCRIPTION
**Pull request by cppmonkey**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
@cppmonkey Thanks for this.  Any chance you could change array() to []  We are trying to convert all the declarations, as code changes are made.

File: includes/discovery/sensors/state/cisco.inc.php [hunk 1]
```diff
@@ -10,31 +10,31 @@
...
-    array('cefcFRUPowerStatusTable','.1.3.6.1.4.1.9.9.117.1.1.2.1.2.','cefcFRUPowerOperStatus','Sensor Name', 'CISCO-ENTITY-FRU-CONTROL-MIB') ,
-    array('cswGlobals','.1.3.6.1.4.1.9.9.500.1.1.3.','cswRingRedundant','Stack Ring - Redundant', 'CISCO-STACKWISE-MIB') ,
-    array('cswSwitchRole','.1.3.6.1.4.1.9.9.500.1.2.1.1.3.','cswSwitchRole','Stack Role - Switch#', 'CISCO-STACKWISE-MIB') ,
-    array('cswSwitchState','.1.3.6.1.4.1.9.9.500.1.2.1.1.6.','cswSwitchState','Stack State - Switch#', 'CISCO-STACKWISE-MIB') ,
-    array('cswStackPortOperStatus','.1.3.6.1.4.1.9.9.500.1.2.2.1.1.','cswStackPortOperStatus','Stack Port Status - ', 'CISCO-STACKWISE-MIB') ,
-    array('cRFCfgRedundancyOperMode','.1.3.6.1.4.1.9.9.176.1.2.14.','cRFCfgRedundancyOperMode','VSS Mode', 'CISCO-RF-MIB') ,
-    array('cRFStatusUnitState','.1.3.6.1.4.1.9.9.176.1.1.2.','cRFStatusUnitState','VSS Device State', 'CISCO-RF-MIB') ,
-    array('cRFStatusPeerUnitState','.1.3.6.1.4.1.9.9.176.1.1.4.','cRFStatusPeerUnitState','VSS Peer State', 'CISCO-RF-MIB')
-);
+$tables = [
```
Comment (cppmonkey) [inline includes/discovery/sensors/state/cisco.inc.php]: This table is nasty... Magic numbers :(
Why is $tablevalue[0] and $tablevalue[2] the same?

**Comment by PipoCanaja** [comment]
Nice and heavy PR ! And best of all, Travis CI is Happy :D

**Comment by cppmonkey** [comment]
Yea, took me a while to get it to be happy.
For a while I couldn't get the snmpsim and --db to work... Blooming amater developers eh!

Could certainly broaden the scope of the PR, but I think I've stretched it enough

**Comment by murrant** [comment]
That was a long read :)

Thanks @cppmonkey 

Hopefully we don't break anything, since not everything has test data.

**Pull request merged**